Gems/ONNX/Code/Source/Clients/ONNXSystemComponent.cpp

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "ONNXSystemComponent.h"
+
+#include "Mnist.h"
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/EditContextConstants.inl>
+#include <AzCore/Serialization/SerializeContext.h>
+
+namespace ONNX
+{
+    void ONNXSystemComponent::SetPrecomputedTimingData(int totalCount, int64_t correctCount, float totalTime, float avgTime)
+    {
+        m_precomputedTimingData->m_totalNumberOfInferences = totalCount;
+        m_precomputedTimingData->m_numberOfCorrectInferences = correctCount;
+        m_precomputedTimingData->m_totalPrecomputedRuntime = totalTime;
+        m_precomputedTimingData->m_averagePrecomputedRuntime = avgTime;
+    }
+
+    PrecomputedTimingData* ONNXSystemComponent::GetPrecomputedTimingData()
+    {
+        return m_precomputedTimingData.get();
+    }
+
+    void ONNXSystemComponent::SetPrecomputedTimingDataCuda(int totalCount, int64_t correctCount, float totalTime, float avgTime)
+    {
+        m_precomputedTimingDataCuda->m_totalNumberOfInferences = totalCount;
+        m_precomputedTimingDataCuda->m_numberOfCorrectInferences = correctCount;
+        m_precomputedTimingDataCuda->m_totalPrecomputedRuntime = totalTime;
+        m_precomputedTimingDataCuda->m_averagePrecomputedRuntime = avgTime;
+    }
+
+    PrecomputedTimingData* ONNXSystemComponent::GetPrecomputedTimingDataCuda()
+    {
+        return m_precomputedTimingDataCuda.get();
+    }
+
+    void ONNXSystemComponent::AddTimingSample(const char* modelName, float inferenceTimeInMilliseconds)
+    {
+        m_timingStats.PushHistogramValue(modelName, inferenceTimeInMilliseconds, AZ::Color::CreateFromRgba(229, 56, 59, 255));
```

Comment:
Does the model have a color member already? if not, let's add that so that each model can be colored differently and the data is coming directly from the model

Reply:
There was no color member in the model - I've added one now.